app/services/ConsignmentExportService.scala

```diff
@@ -0,0 +1,33 @@
+package services
+
+import java.util.UUID
+
+import com.nimbusds.oauth2.sdk.token.BearerAccessToken
+import configuration.GraphQLConfiguration
+import ApiErrorHandling._
+import javax.inject.Inject
+import play.api.Configuration
+import play.api.libs.ws.{WSClient, WSResponse}
+import graphql.codegen.UpdateTransferInitiated.updateTransferInitiated._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ConsignmentExportService @Inject()(val ws: WSClient, val configuration: Configuration, graphQLConfiguration: GraphQLConfiguration)
+                                        (implicit val executionContext: ExecutionContext) {
+
+  def updateTransferInititated(consignmentId: UUID, token: BearerAccessToken): Future[Boolean] = {
+    val client = graphQLConfiguration.getClient[Data, Variables]()
+    sendApiRequest(client, document, token, Variables(consignmentId))
+      .map(d => d.updateTransferInitiated.isDefined)
+      .recover(_ => false)
```

Comment:
Can you make sure that any exceptions from failed futures and GraphQL errors are logged?

Reply:
The sendApiRequest method checks for errors from the GraphQL response and throws them as exceptions if there are any so any errors at all will be contained within the failed future. I've added a logging line into the recover block to log the error and this should log graphql errors and any network errors.